### DISCUSSION TITLE
mapping two source props to one target prop

### DISCUSSION THREAD

**Discussion by senal** [riok/mapperly#727] [Q&A] [answered]
Hi,

I have a requirement to bind/join two source property values to one `target` property.

source 

     public class Car 
     {
         public string Brand {get; set;}
         public string Model {get; set;}
     }

target

     public class CarDto 
     {
         public string Make {get; set;}
     }

I need to join the value of `Brand and Model` properties of the `Car` object to `Make` property of `CarDto`

Appreciate your help.

**Comment by TimothyMakkison** [comment, marked as answer]
Hey, thanks for using Mapperly. It looks like [before/after](https://mapperly.riok.app/docs/configuration/before-after-map/) mapping could work.

```C#
[Mapper]
public static partial class Mapper
{
    // tell mapperly it doesn't need to resolve Make, we will do it ourselves
    [MapperIgnoreTarget(nameof(CarDto.Make))]
    private static partial CarDto InternalMapCarToDto(Car car);

    public static CarDto MapCarToDto(Car car)
    {
        var target = InternalMapCarToDto(car);
        target.Make = $"Merged {car.Brand} + {car.Model}";
        return target;
    }
}
```

**Reply by senal** [reply]
Thanks @TimothyMakkison ,
But I'm getting this error. Any thoughts?

![image](https://github.com/riok/mapperly/assets/2841066/810e8f9b-159b-4db7-8f8c-ac8d565e80ce)

My implementation is as below:


     [Mapper]
     public static partial class CarMapper
     {
         [MapperIgnoreTarget(nameof(CarDto.Make))]
         private static partial CarDto InternalMapDtoToCar(Car car);

         public static CarDto MapCarToDto(Car car)
         {
            var target = MapCarToDto(car);
            target.Make = $"{car.Brand} {car.Model}";
            return target;
         }
     }

**Comment by senal** [comment]
@TimothyMakkison ,

I think  

    
    var target = MapCarToDto(car);


has to change as below


    
    var target = InternalMapDtoToCar(car);


That fix the exception.

**Reply by TimothyMakkison** [reply]
Oh yeah my bad. I wasn't using an ide so I didnt catch that. Sorry 🙏
